lib/common-types.ts

```diff
@@ -1 +1,13 @@
+import { notStrictEqual } from 'assert'
+
 export type Dictionary<T = any> = { [key: string]: T }
+
+export type NotEmptyArray<T = any> = [T, ...T[]]
+
```

Comment:
is the idea that these types aren't specific to yargs data structures?

Reply:
Yes. The main reason for this is `split()` signature is not precise enough: it returns `string[]` as if the first argument could be undefined, which is not the case. Its real signature should be `[string, ...string[]]`.

When you are destructuring the results of `split()`, for example:
```ts
const [cmd, ...args] = something.split();
```
then cmd is `string | undefined` when it should be `string`.

Reply:
To answer more precisely your question: `common-types` was at first the place to put types several modules would use, but at the moment is mostly used by types which I think should be typescript standards, but are not yet implemented.